docs/deploying.rst

```diff
@@ -0,0 +1,328 @@
+be inaccessible without that option.
+
+If you create a ``setup.py`` for your project (see `setuptools`_), the ``cp``
+step can be replaced with ``pip install .``. Your ``setup.py`` should declare
+the ``fairydust`` dependency, also making redundant the manual installation of
+that package in the steps above. Note that it is not possible to use ``python
+setup.py develop`` or ``pip install -e .`` instead of ``pip install .`` because
+the former two do not copy the source files but create an ``.egg-link`` file
+instead, which Toil is not able to hot-deploy. Similarly, ``python setup.py
+install`` does not work either because it installs the project as a Python Egg
+(a ``.egg`` file), which is not supported by Toil although that may `change`_
+in the future. You might be tempted to prevent the installation of the ``.egg``
+by passing ``--single-version-externally-managed`` to ``setup.py install`` but
+that would also disable the automatic installation of your project's
+dependencies.
+
+.. _setuptools: http://setuptools.readthedocs.io/en/latest/index.html
+.. _change: https://github.com/BD2KGenomics/toil/issues/1367
+
+If you publish your project to PyPI, others will be able to install it on their
+leader using pip, provided they 1) already installed Toil on the leader and
+workers nodes and 2) use a virtualenv created with ``--system-site-packages``::
+
+   $ virtualenv --system-site-packages my-project
+   $ . my-project/bin/activate
+   $ pip install my-project
+   $ python -m workflow.main --batchSystem=mesos …
+
+Relying on shared filesystems
+-----------------------------
+
+Bare-metal clusters typically mount a shared file system like NFS on each node.
+If every node has that file system mounted at the same path, you can place your
+project on that shared filesystem and run your user script from there.
+Additionally, you can clone the Toil source tree into a directory on that
+shared file system and you won't even need to install Toil on every worker. Be
+sure to add both your project directory and the Toil clone to ``PYTHONPATH``. Toil
+replicates ``PYTHONPATH`` from the leader to every worker.
+
+.. _deploying_toil:
+
+Deploying Toil
+--------------
+
+We've looked at various ways of installing your workflow on the leader such
+that Toil can replicate it to the workers and load the job definitions there.
+But what about Toil itself? Unless you are running your workflow in single
+machine mode (the default) or on a cluster where every node mounts a shared
+file system at the same path, Toil somehow needs to be made available on each
+worker. Unfortunately, hot-deployment only works for the user script/module and
+its dependencies, not for Toil itself. Generally speaking, you or your admin
+will need to manually :ref:`install <installation>` Toil on every cluster node
+you intend to run Toil jobs on.
+
+The Toil team is eagerly working to ameliorate this. Toil 3.5.0 will contain
```

Comment:
This paragraph and the one below are now outdated, can you update this? This mostly looks like this is a duplicate of the 'running on cloud environment' docs so you can either delete this or replace most of it to a link to that section.